### PR TITLE
Removing the content of VSTS Cleanup Agent.

### PR DESCRIPTION
**Pull request by JohnTortugo**
This is part of the changes to address this issue: https://github.com/dotnet/core-eng/issues/2964

### CODE REVIEW

File: src/Microsoft.DotNet.Build.Tasks/CleanupVSTSAgent.cs [hunk 1]
```diff
@@ -1,645 +1,11 @@
...
-                    {
-                        Log.LogMessage($"Will retry again in {SleepTimeInMilliseconds} ms");
-                        await System.Threading.Tasks.Task.Delay(SleepTimeInMilliseconds.Value);
-                        return await CleanupDirectoryAsync(directory, attempts).ConfigureAwait(false);
-                    }
-                }
-            }
-            Log.LogMessage("Failed to cleanup.");
+            Log.LogMessage($"This BuildTask has been deprecated.");
             return false;
```
Comment (MattGal) [inline src/Microsoft.DotNet.Build.Tasks/CleanupVSTSAgent.cs]: There's no reason to fail over a deprecated task.  Also might mention why (in favor of maintenance jobs)

File: src/Microsoft.DotNet.Build.Tasks/CleanupVSTSAgent.cs [hunk 2]
```diff
@@ -1,645 +1,11 @@
...
-        private static readonly int s_DefaultSleepTime = 2000;
-        private DateTime _timerStarted;
-
-        private HashSet<string> _protectedDirectories = new HashSet<string>();
-        private HashSet<string> _additionalCleanupDirectories = new HashSet<string>();
-        private Dictionary<string, bool> _availableUnixCommands = new Dictionary<string, bool>();
-
         public override bool Execute()
         {
-            string entryLocation = System.Reflection.Assembly.GetEntryAssembly().Location;
```
Comment (MattGal) [inline src/Microsoft.DotNet.Build.Tasks/CleanupVSTSAgent.cs]: Suggestion: Obsolete messages should be warnings, not messages.

File: src/Microsoft.DotNet.Build.Tasks/CleanupVSTSAgent.cs (hunk 1 above)
Comment (JohnTortugo) [reply src/Microsoft.DotNet.Build.Tasks/CleanupVSTSAgent.cs]: Thanks. Fixed.

File: src/Microsoft.DotNet.Build.Tasks/CleanupVSTSAgent.cs [hunk 3]
```diff
@@ -1,645 +1,11 @@
...
-
 namespace Microsoft.DotNet.Build.Tasks
 {
     public class CleanupVSTSAgent : BuildTask
     {
-        public bool Clean { get; set; }
-
-        public bool Report { get; set; }
-
-        [Required]
```
Comment (weshaggard) [inline src/Microsoft.DotNet.Build.Tasks/CleanupVSTSAgent.cs]: If there are any targets files calling these properties they will start to fail after this change. If we want to make it less impact-ful you should keep the properties.

**Comment by MattGal** [comment]
Ah good point;  @JohnTortugo I was mistakenly assuming extra properties were ignored. Should fix this ASAP in case a team which has not had the usage removed picks up new buildtools.

File: src/Microsoft.DotNet.Build.Tasks/CleanupVSTSAgent.cs (hunk 3 above)
Comment (JohnTortugo) [reply src/Microsoft.DotNet.Build.Tasks/CleanupVSTSAgent.cs]: Added new PR: https://github.com/dotnet/buildtools/pull/1998

**Pull request merged**